src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -49,45 +61,123 @@ struct KeyValueViewModel {
+                needDecodingInMethod = false
+            case .date,
+                 .dateTime,
+                 .unixTime:
+                self.value = value
+                self.constantValue = "\"\(constantValue)\""
+                keyValueType = .dateFromConstant
+            case .byteArray:
+                self.value = value
+                self.constantValue = "\"\(constantValue)\""
+                keyValueType = .byteArrayFromConstant
+            case .number:
+                self.value = value
+                self.constantValue = "Double(\(constantValue))"
+            default:
+                self.value = value
+                self.constantValue = constantValue
+            }
+            self.needDecodingInMethod = needDecodingInMethod
+        } else if let signatureParameter = operation.signatureParameter(for: name) {
             self.optional = !signatureParameter.required
-            let swiftType = signatureParameter.schema.swiftType(optional: optional)
-            if swiftType.starts(with: "String") {
-                self.value = param.name
-            } else if swiftType.starts(with: "Date") {
-                self.value = "String(describing:\(param.name), format: Date.Format.iso8601)"
-            } else {
-                // Convert into String in generated code
-                self.value = "String(\(param.name))"
+            self.constantValue = nil
+
+            // value is referring a signature parameter, no need to wrap as String
+            self.value = convertValueToStringInSwift(
+                type: signatureParameter.schema.type,
+                value: name
+            )
+
+            // if parameter is from method signature (not from option) and type is date or byteArray,
+            // add decoding logic to string in the method and specify the right decoding strategy
+            switch signatureParameter.schema.type {
+            case .date,
+                 .unixTime,
+                 .dateTime:
+                keyValueType = signatureParameter.required ? .dateFromSignature : .dateFromConstant
+                self.needDecodingInMethod = true
+            case .byteArray:
+                keyValueType = signatureParameter.required ? .byteArrayFromSignature : .byteArrayFromConstant
+                self.needDecodingInMethod = true
+            default:
+                self.needDecodingInMethod = false
             }
+        } else if param.implementation == ImplementationLocation.client {
+            self.value = "client.\(name)"
+            self.optional = false
+            self.constantValue = nil
+            self.needDecodingInMethod = false
         } else {
             self.value = ""
             self.optional = false
-            self.paramName = nil
+            self.constantValue = nil
+            self.needDecodingInMethod = false
         }
+
+        self.key = name
+        self.strategy = keyValueType.rawValue
     }
```

Comment:
This seems like a very large initializer. It might make more sense to split this into different convenience intializers.

Reply:
Broke up into small init() for each scenario